### PR TITLE
Implement Ord instance for PreciseDuration

### LINKED ISSUE #22: PreciseDuration Eq and Ord instances are wrong
They should normalize to nanos and compare.

### PR DESCRIPTION
**Pull request by paulyoung**
Fixes #22

### CODE REVIEW

File: src/Data/Time/PreciseDuration.purs [hunk 1]
```diff
@@ -12,17 +12,47 @@ module Data.Time.PreciseDuration
...
+  -- Nanoseconds must be integral, see smart constructor in Data.Time.PreciseDuration.
+  | Microseconds Decimal
+  | Milliseconds Decimal
+  | Seconds Decimal
+  | Minutes Decimal
+  | Hours Decimal
+  | Days Decimal
+  | Weeks Decimal
+
+derive instance eqPreciseDuration :: Eq PreciseDuration
```
Comment (natefaubion) [inline src/Data/Time/PreciseDuration.purs]: Can't derive Eq.

Comment (natefaubion) [reply src/Data/Time/PreciseDuration.purs]: But I suppose you could use `compare a b == EQ`

File: src/Data/Time/PreciseDuration.purs [hunk 2]
```diff
@@ -12,17 +12,47 @@ module Data.Time.PreciseDuration
...
+
+instance ordPreciseDuration :: Ord PreciseDuration where
+  compare x y = compare (toNanoseconds x) (toNanoseconds y)
```
Comment (natefaubion) [inline src/Data/Time/PreciseDuration.purs]: You could also do something like
```purescript
compare (Weeks a) (Weeks b) = compare a b
compare (Weeks a) b | Weeks b' <- toWeeks b = compare a b
...
```
Not sure if that will just be any more efficient though.

**Comment by paulyoung** [comment]
Stack overflow 😢

**Comment by paulyoung** [comment]
When I run `pulp test` locally I see this:

```
var compare = function (dict) {
                       ^

RangeError: Maximum call stack size exceeded
```

`output/Data.Time.PreciseDuration/index.js` contains:

```js
var ordPreciseDuration = new Data_Ord.Ord(function () {
    return eqPreciseDuration;
}, function (x) {
    return function (y) {
        return Data_Ord.compare(ordPreciseDuration)(toNanoseconds(x))(toNanoseconds(y));
    };
});
var eqPreciseDuration = new Data_Eq.Eq(function (x) {
    return function (y) {
        return Data_Eq.eq(Data_Ordering.eqOrdering)(Data_Ord.compare(ordPreciseDuration)(x)(y))(Data_Ordering.EQ.value);
    };
});
```

**Pull request merged**